src/toxcore/error.rs

```diff
@@ -0,0 +1,122 @@
+/*! Error definitions
```

Comment:
No need to move it into toxcore/error.rs if you use it only in dht/daemon_state.rs

Reply:
How about all error definitions is in this error.rs module. Packet errors, codec errors, Tcp, Onion,...

Reply:
@kurnevsky ?

Reply:
It seems @NamsooCho means that this error can be common for all deserialization cases. But it can't be applicable to `get_payloads` methods unless we add a decryption error here. And anyway it has to be named properly in this case, not just `Error`. If it's up to me I'd leave it in `daemon_state` module and later unify if we need the same errors anywhere else.

Reply:
done